File: TokanPages.Backend/TokanPages.Backend.Domain/Entities/User/UserNote.cs
using System.ComponentModel.DataAnnotations;
using System.Diagnostics.CodeAnalysis;
using TokanPages.Backend.Domain.Contracts;

namespace TokanPages.Backend.Domain.Entities.User;

[ExcludeFromCodeCoverage]
public class UserNote : Entity<Guid>, IAuditable
{
    public Guid UserId { get; set; }
    [Required]
    public string Note { get; set; }
    public Guid CreatedBy { get; set; }
    public DateTime CreatedAt { get; set; }
    public Guid? ModifiedBy { get; set; }
    public DateTime? ModifiedAt { get; set; }

    /* Navigation properties */
    public Users Users { get; set; }
}

File: TokanPages.ClientApp/src/Store/Defaults/Users/userNote.ts
import { OperationStatus } from "../../../Shared/enums";
import { UserNoteState } from "../../../Store/States";

export const UserNote: UserNoteState = {
    status: OperationStatus.notStarted,
    response: {
        id: "",
        note: "",
        createdBy: "",
        createdAt: "",
        modifiedBy: "",
        modifiedAt: ""
    }
} 
